processor/src/main/java/org/mapstruct/ap/internal/conversion/Conversions.java

```diff
@@ -324,10 +328,18 @@ private void register(String sourceTypeName, Class<?> targetClass, ConversionPro
     }
 
     public ConversionProvider getConversion(Type sourceType, Type targetType) {
-        if ( sourceType.isEnumType() && targetType.equals( stringType ) ) {
+        if ( sourceType.isEnumType() &&
+                ( targetType.equals( stringType ) ||
+                  targetType.equals( integerType ) ||
+                  targetType.getBoxedEquivalent().equals( integerType ) )
```

Comment:
When doing `getBoxedEquivalent` if the target type is not primitive it returns itself. This means that we can just do:

```suggestion
                  targetType.getBoxedEquivalent().equals( integerType ) )
```